apps/io-services-cms-webapp/src/webservice/controllers/get-service-topics.ts

```diff
@@ -0,0 +1,96 @@
+import { Context } from "@azure/functions";
+import {
+  AzureApiAuthMiddleware,
+  IAzureApiAuthorization,
+  UserGroup,
+} from "@pagopa/io-functions-commons/dist/src/utils/middlewares/azure_api_auth";
+import { ContextMiddleware } from "@pagopa/io-functions-commons/dist/src/utils/middlewares/context_middleware";
+import {
+  withRequestMiddlewares,
+  wrapRequestHandler,
+} from "@pagopa/io-functions-commons/dist/src/utils/request_middleware";
+import { initAppInsights } from "@pagopa/ts-commons/lib/appinsights";
+import {
+  IResponseSuccessJson,
+  ResponseErrorInternal,
+  ResponseSuccessJson,
+} from "@pagopa/ts-commons/lib/responses";
+import * as O from "fp-ts/lib/Option";
+import * as TE from "fp-ts/lib/TaskEither";
+import { flow, pipe } from "fp-ts/lib/function";
+
+import { ServiceTopicList } from "../../generated/api/ServiceTopicList";
+import {
+  EventNameEnum,
+  trackEventOnResponseOK,
+} from "../../utils/applicationinsight";
+import { ErrorResponseTypes, getLogger } from "../../utils/logger";
+import { ServiceTopicDao } from "../../utils/service-topic-dao";
+
+const logPrefix = "GetServiceTopicsHandler";
+
+type HandlerResponseTypes =
+  | IResponseSuccessJson<ServiceTopicList>
+  | ErrorResponseTypes;
+
+type GetServiceTopicsHandler = (
+  context: Context,
+  auth: IAzureApiAuthorization
+) => Promise<HandlerResponseTypes>;
+
+type Dependencies = {
+  // A store od ServiceLifecycle objects
+  telemetryClient: ReturnType<typeof initAppInsights>;
+  serviceTopicDao: ServiceTopicDao;
+};
+
+export const makeGetServiceTopicsHandler =
+  ({
+    telemetryClient,
+    serviceTopicDao,
+  }: Dependencies): GetServiceTopicsHandler =>
+  (context, auth) =>
+    pipe(
+      serviceTopicDao.findAllNotDeletedTopics(),
+      TE.map(
+        flow(
+          O.fold(
+            // QUESTION: when no topics are found on DB should we return an empty list or an error 404?
```

Comment:
**note:** in general, for rest services like "findAll", "search" or something like that, I prefer to return an empty list rather that a 404, but honestly I think both solutions are valid, so is up to us to choose the one we prefer

Reply:
I agree! It should return an empty lists, not a 404.

Ideally REST endpoints that returns array should also wrap theme inside object (in other words, an object should always be the root data stracture), this approach lets you add metadata (ie. legth, page, ...) and is future proof. Something like:

```
{
  items: [ ... ],
}
```

not need for this PR anyway.